benchmarks/modules/utils.py

```diff
@@ -267,10 +269,22 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        cmd_compiler_name = 'from spack import environment; print(environment.active_environment().spec_lists["specs"].specs[0].compiler.name)'
+        cmd_compiler_version = 'from spack import environment; environment.active_environment().spec_lists["specs"].specs[0].compiler.versions[0]'
```

Comment:
There can be more than one package in the spec, so instead of the `[0]` element, we should write out a dictionary for each package. Something like `... | {'compiler_name_package1':'name1', 'compiler_name_package2':'name2', ...} | 'compiler_version_package1':'version1', 'compiler_version_package2':'version2', ...} | <other parts of the spec> ...| The postprocessing would then have to be minimally modified to be able to parse this, and create a separate column for each key of each dictionary.